internal/pkg/service/stream/storage/node/coordinator/slicerotation/operator.go

```diff
@@ -248,38 +256,62 @@ func (o *operator) rotateSlice(ctx context.Context, slice *sliceData) {
 
 	// Rotate slice
 	o.logger.Infof(ctx, "rotating slice for upload: %s", cause)
+	err = o.rotateSliceWithState(ctx, slice)
+	if err != nil {
+		return
+	}
+
+	o.logger.Info(ctx, "successfully rotated slice")
+	// Prevents other processing, if the entity has been modified.
+	// It takes a while to watch stream send the updated state back.
+	slice.Processed = true
+}
 
+func (o *operator) rotateSliceWithState(ctx context.Context, slice *sliceData) error {
 	// Lock all file operations in the sink
 	lock, unlock := sinklock.LockSinkFileOperations(ctx, o.locks, o.logger, slice.SliceKey.SinkKey)
 	if unlock == nil {
-		return
+		return errors.New("unable to lock file")
 	}
 	defer unlock()
 
 	// Rotate slice
-	err = o.storage.Slice().Rotate(slice.SliceKey, o.clock.Now()).RequireLock(lock).Do(ctx).Err()
+	err := o.storage.Slice().Rotate(slice.SliceKey, o.clock.Now()).RequireLock(lock).Do(ctx).Err()
 	// Handle error
 	if err != nil {
 		o.logger.Errorf(ctx, "cannot rotate slice: %s", err)
 
 		// Increment retry delay
-		err := o.storage.Slice().IncrementRetryAttempt(slice.SliceKey, o.clock.Now(), err.Error()).RequireLock(lock).Do(ctx).Err()
-		if err != nil {
+		iErr := o.storage.Slice().IncrementRetryAttempt(slice.SliceKey, o.clock.Now(), err.Error()).RequireLock(lock).Do(ctx).Err()
+		if iErr != nil {
 			o.logger.Errorf(ctx, "cannot increment slice retry: %s", err)
-			return
+			return iErr
 		}
 	}
 
-	// Prevents other processing, if the entity has been modified.
-	// It takes a while to watch stream send the updated state back.
-	slice.Processed = true
+	return nil
 }
 
 func (o *operator) closeSlice(ctx context.Context, slice *sliceData) {
+	// Rotate slice
+	o.logger.Infof(ctx, "closing slice")
+
 	// Wait until no source node is using the slice
 	ctx, cancel := context.WithTimeout(context.WithoutCancel(ctx), o.config.SliceCloseTimeout.Duration())
 	defer cancel()
 
+	err := o.switchSliceToImporting(ctx, slice)
+	if err != nil {
+		return
+	}
```

Comment:
I think you mean this code.
https://github.com/keboola/keboola-as-code/blob/fix-node-operator-logging-consolidation/internal/pkg/service/stream/storage/node/coordinator/slicerotation/operator.go#L331-L341
        If you can see, we can receive error when switching state to `uploading`.
In case we get error of this switching operation we try to increment the RetryAttempt. However, I use another error `rErr` to differentiate between these two.
1. In case we cannot increment attempt, it is  `!processed`
2. In case we can switch the increment attempt the function returns `nil` which set's it to `processed`.
3. In case we have no error is set's it to `processed`.

I've had different refactor in place but I decided to not go with it as your code was approved.

Reply:
I changed `iErr` -> `rErr`, just cosmetic thing

Reply:
should be fixed https://github.com/keboola/keboola-as-code/pull/1960/commits/454749fc6caec2df4d38fd6cc05a96390933d3d4